### PR TITLE
New App: msteamsstatus

### PR DESCRIPTION
**Pull request by schumatt**
# Description
Initial commit for msteamsstatus app. Please insert appropriate hash of client ID & secret for Tidbyt's AAD tenant on lines 29 and 30.

# Copilot

### Summary
📄📝📱

This pull request adds a new app called `msteamsstatus` to the Tidbyt community repository. The app shows the user's Microsoft Teams presence on a Tidbyt device, and includes a manifest file and a README file.

> _`msteamsstatus` app_
> _manifest and README files_
> _autumn leaves of code_

### Walkthrough
* Create a new app called msteamsstatus that shows the user's Microsoft Teams presence on a Tidbyt device ([link](https://github.com/tidbyt/community/pull/1864/files?diff=unified&w=0#diff-ccbf659ce40cb864da62e9de98f1c53ba98e361cf91c8682c9320a89dd88a16bR1-R8), [link](https://github.com/tidbyt/community/pull/1864/files?diff=unified&w=0#diff-cdd3dcc305a851fc38d911c6a3357e87f4489252466187982afbceb499247d44L1-R8))
* Add a manifest file that defines the app's metadata and package name ([link](https://github.com/tidbyt/community/pull/1864/files?diff=unified&w=0#diff-ccbf659ce40cb864da62e9de98f1c53ba98e361cf91c8682c9320a89dd88a16bR1-R8))
* Add a README file that explains the app's purpose, features, and credits ([link](https://github.com/tidbyt/community/pull/1864/files?diff=unified&w=0#diff-cdd3dcc305a851fc38d911c6a3357e87f4489252466187982afbceb499247d44L1-R8))

### CODE REVIEW

**Comment by schumatt** [comment]
I have read the CLA Document and I hereby sign the CLA

File: apps/msteamsstatus/ms_teams_status.star [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        if response.status_code != 200:
+            fail("Refresh of Access Token failed with Status Code: %d - %s" % (response.status_code, response.body()))
+
+        response_json = response.json()
+
+        # TODO: Determine if this cache call can be converted to the new HTTP cache.
+        cache.set(
+            response_json["refresh_token"],
+            response_json["access_token"],
+            ttl_seconds = int(response_json["expires_in"] - 30),
```
Comment (noahpodgurski) [inline apps/msteamsstatus/ms_teams_status.star:343]:
```suggestion
            ttl_seconds = int(response_json["expires_in"]) - 30,
```

**Comment by schumatt** [comment]
@noahpodgurski Updated with your suggested change. Thanks.

File: apps/msteamsstatus/ms_teams_status.star [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+        fail("token request failed with status code: %d - %s" %
+             (response.status_code, response.body()))
+
+    response_json = response.json()
+
+    # TODO: Determine if this cache call can be converted to the new HTTP cache.
+    cache.set(
+        response_json["refresh_token"],
+        response_json["access_token"],
+        ttl_seconds = int(response_json["expires_in"] - 30),
```
Comment (noahpodgurski) [inline apps/msteamsstatus/ms_teams_status.star:379]:
```suggestion
        ttl_seconds = int(response_json["expires_in"]) - 30,
```

Again here

**Comment by schumatt** [comment]
Hey all - any updates here? Is there something I need to do to move this forward?

File: apps/msteamsstatus/ms_teams_status.star [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+defaultUserDisplayName = "Not Signed In"
+defaultAvailability = "PresenceUnknown"
+defaultActivity = "PresenceUnknown"
+defaultStatusMessage = ""
+
+devClientID = "1"
+devClientSecret = "1"
+defaultTenantID = "common"
+
+prodClientIDHash = ""
```
Comment (matslina) [inline apps/msteamsstatus/ms_teams_status.star:29]:
```suggestion
prodClientIDHash = "AV6+xWcE16fH0o1Om+eSnLeFELBiEuFy51tl1mnq43AJq2u5Y0HNcv4/omDaUTSFhm7o+P3OKaKhyYITqkOC+yd0wOIV8n6hbGxNwJpx/3a5nPlVTKGAWawf4AnsbK6GKjlYdC9YFWsIIBlA6pUgs5ck4w1DCbqEBXZlXdbS6qsVrxWmrA193Q4x"
```

File: apps/msteamsstatus/ms_teams_status.star [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+prodClientSecretHash = ""
```
Comment (matslina) [inline apps/msteamsstatus/ms_teams_status.star:30]:
```suggestion
prodClientSecretHash = "AV6+xWcEHLnCoQThNTWlS0tGWq92/FBTALw47B3BPSygpsIT5PThPtSjpdy6SR9x5M/NC7qtyNWyUgdKsbtxpYS1lSSbaF+RB3VCQEuiMS8MHoyHCg2Jh6BgLXTmnZJoNZyJQgIu9FYqH6MNFCE+uoJWPYp2ceENHhzRKbP5tNEUhrfrPSUv3J9qPOc+ig=="
```

**Comment by matslina** [comment]
Sorry for the delay. Just a pain to click through all the microsoft things to get credentials out. Let's try with the creds from outlookcalendar for now.

File: apps/msteamsstatus/ms_teams_status.star [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+defaultUserDisplayName = "Not Signed In"
+defaultAvailability = "PresenceUnknown"
+defaultActivity = "PresenceUnknown"
+defaultStatusMessage = ""
+
+devClientID = "1"
+devClientSecret = "1"
+defaultTenantID = "common"
+
+prodClientIDHash = "AV6+xWcE16fH0o1Om+eSnLeFELBiEuFy51tl1mnq43AJq2u5Y0HNcv4/omDaUTSFhm7o+P3OKaKhyYITqkOC+yd0wOIV8n6hbGxNwJpx/3a5nPlVTKGAWawf4AnsbK6GKjlYdC9YFWsIIBlA6pUgs5ck4w1DCbqEBXZlXdbS6qsVrxWmrA193Q4x"
```
Comment (matslina) [inline apps/msteamsstatus/ms_teams_status.star:29]:
```suggestion
prodClientIDHash = "AV6+xWcEdXjnrJSgScSB3jcsEfzpcOdwTaf7JDGgOCZRpunzZe6Lfh9kM1hPckFHWb4xLgP6odaUSmusWaahx1/Y5mQ60LyAHpl3IrJsfyma67ANw87LfZ1zCPJS5q+3lJA6I6FFq4poy3Hzjagv8itdOY6BNpVYf9BwgmS/ZCq0uAVU/ujpO5KF"
```

File: apps/msteamsstatus/ms_teams_status.star [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+prodClientSecretHash = "AV6+xWcEHLnCoQThNTWlS0tGWq92/FBTALw47B3BPSygpsIT5PThPtSjpdy6SR9x5M/NC7qtyNWyUgdKsbtxpYS1lSSbaF+RB3VCQEuiMS8MHoyHCg2Jh6BgLXTmnZJoNZyJQgIu9FYqH6MNFCE+uoJWPYp2ceENHhzRKbP5tNEUhrfrPSUv3J9qPOc+ig=="
```
Comment (matslina) [inline apps/msteamsstatus/ms_teams_status.star:30]:
```suggestion
prodClientSecretHash = "AV6+xWcESxIErxRljgKqjXhjsv7zgrV0cFLF0hYUNAhWZthU12DQliI+R7lP7PUk7yrzCBEGci/s1unk/GSjjKiPfRkkwLSOGWtUIpyEcRcQ0dqy68uAUVlRqeALq02sWnKO91djJCyp+vgePYcj368ofqhL3Fh3F+NZftvtKyr7r1JloNN6/8ZiFceU8A=="
```

**Pull request merged**